lisa/tools/cargo.py

```diff
@@ -56,15 +77,42 @@ def _install(self) -> bool:
                 target=f"{home_dir}/.cargo/bin/cargo",
                 link="/usr/local/bin/cargo",
             )
+            ln.create_link(
+                is_symbolic=True,
+                target=f"{home_dir}/.cargo/bin/rustup",
+                link="/usr/local/bin/rustup",
+            )
         else:
-            raise UnsupportedDistroException(os)
-        return self._check_exists()
+            raise UnsupportedDistroException(node_os)
+        is_installed = self._check_exists()
+
+        # Point cargo from stable toolchain of rust after installation
+        if is_installed:
+            self.node.execute("rustup default stable", shell=True)
+            self.toolchain = self.__get_rust_toolchain()
+            self._log.debug(f"Rust toolchain: {self.toolchain}")
+            self.command = "".join(
+                [
+                    f"{home_dir}/.rustup/toolchains/",
+                    f"{self.toolchain}/bin/cargo",
+                ]
+            )
```

Comment:
`join` is not needed. Python will concatenate the strings if you do this:
```suggestion
            self.command = (
                    f"{home_dir}/.rustup/toolchains/"
                    f"{self. Toolchain}/bin/cargo"
            )
```

Reply:
@anirudhrb i have done it earlier that way only but flak8 validation is having issue with this way of string. It would say **BLK100 would make change** at this line. That is why i did in this way.

Reply:
What happens when you try to autocorrect that error?

Reply:
@anirudhrb  It wont show any error with flak8 validation in code editor but with pipeline it does. It wont mention anything in detail with pipeline instance, just **BLK100 would make change**

Reply:
@anirudhrb As we discussed, i ran the nox manually with python 3.8 locally and it is passing with flak8. It is failing with pipeline instance only when PR branch push. Please find below log.

nox > Session black was successful.
nox > Running session isort
nox > Creating virtual environment (virtualenv) using python.exe in .nox\isort
nox > python -m pip install 'isort ~= 5.12.0'
nox > isort .
Skipped 5 files
nox > Session isort was successful.
nox > Running session flake8
nox > Creating virtual environment (virtualenv) using python.exe in .nox\flake8
nox > python -m pip install 'black ~= 23.1.0' 'flake8 ~= 6.0.0' Flake8-pyproject 'flake8-black >= 0.3.2' 'flake8-bugbear ~= 23.2.13' 'flake8-isort >= 4.1.1' 'pep8-naming >= 0.12.1' 'isort ~= 5.12.0'
nox > flake8
**nox > Session flake8 was successful.**

Reply:
My guess is that black is complaining about some small formatting issue and not about the whole idea of using implicit f-string continuation. I believe there are other places in the code that do the same thing. You should try to get to the bottom of this.

This is not a blocker though.

Reply:
I removed join and used f-string. Earlier it was character length limit issue which leads to this. It is resolved now.